core/src/main/java/org/mapstruct/Mapping.java

```diff
@@ -281,6 +281,31 @@
      */
     String defaultExpression() default "";
 
+    /**
+     * A condition {@link String} based on which the specified target property is to be set.
+     * <p>
+     * An example mapping would lookg like this
+     * <pre><code>
+     * &#64;Mapping(
+     *     target = "someProp",
+     *     condition = "conditionMethodName"
+     * )
+     * </code></pre>
+     * <p>
+     * will cause the following target property assignment to be generated:
+     * <p>
+     * {@code
+     *   if (conditionMethodName(s)) {
+     *      targetBean.setSomeProp( new TimeAndFormat( s.getTime(), s.getFormat() ) )}
+     *   }
+     * <p>
+     * The condition method needs to be defined as follows:
+     * {@code boolean someName(SourceType source); }
+     *
+     * @return The name of the condition method to be evaluated when trting to do the given target mapping
+     */
+    String condition() default "";
```

Comment:
The example that you are using from the `defaultExpression` is not something valid for `condition`.

I would suggest the following:

Example:

```
@Mapping(
    target = "name",
    condition = "isNotBlank"
)
```

generates:

```
if ( isNotBlank( source.getName() ) ) {
    target.setName( source.getName() );
}
```

And the condition method should be defined as:

```
boolean isNotBlank(String value);
```

Reply:
I'm not sure that this makes it really flexible:
target="name", condition="isNotBlack" always assumes that the condition will be based on the source's corresponding field but this might not be the case. 

It would be much more flexible if the condition method is of the form:
boolean someName(SourceObject type)

Let me know what do you think?

Reply:
I can follow both ideas. I like having the flexibility @Desislav-Petrov proposes with getting the whole source object to do some checks on this one.
But keeping it strict to the value to which the check is assigned as @filiphr proposes makes it easier to reuse the condition methods. Sticking to the `isNotBlank` example: This could be used for all String source values, getting the whole source type will make it more complicated to reuse it for other source values.

One idea to support both: Default to @filiphr proposals and if there is an (not annotated) argument for the method this will be the source value.
And also support if an argument was annotated with `@MappingSource` the source object will be used (do we already have an annotation like this? tbh I dont remember, just can find `@MappingTarget` that maybe could also be supported?!)